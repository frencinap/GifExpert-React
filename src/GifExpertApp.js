import React, { useState } from "react"
import { AddCategory } from "./components/AddCategory"
import { GifGrid } from "./components/GifGrid"
import { ScrollButton } from "./components/ScrollButton"

// vamos a agregar una categoría nueva para trabajarla en el testsuite
//agregaremos defaultcategories en vez de un string vacío
export const GifExpertApp = ( { defaultCategories = [] } ) => {

    const [ categories, setCategories ] = useState( defaultCategories )

    return (
        <>
            <h2> GifExpert App</h2> 
            <AddCategory setCategories={ setCategories } />
            <hr />
            <ol>
                {
                    categories.map( category => (
                        <GifGrid
                            category={category}
                            key={category}
                        />
                    ))
                }    
            </ol>
            <ScrollButton/>
        </>
    )
}