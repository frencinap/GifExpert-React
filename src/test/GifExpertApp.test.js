import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Pruebas en <GifExpertApp /> ', () => {
    
    test('Hacer match con el snapshot', () => {

        const wrapper = shallow( <GifExpertApp/>)  
        expect( wrapper ).toMatchSnapshot()
        
    });

    test('Debe de mostrar una lista de categorÃ­as', () => {

        const categories = [' Eren ', ' Batman ']
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/>)
        // en este caso deberian haber 2 GifGrids con las categoires definidas arriba
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )  
        
    });
});