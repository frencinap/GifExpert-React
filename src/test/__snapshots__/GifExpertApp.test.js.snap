// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp />  Hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
     GifExpert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
  <ScrollButton />
</Fragment>
`;
